### PR TITLE
[Enhancement] remove recursive dir iterator

### PR DESCRIPTION
**Pull request by JehandadKhan**
#1185 
The recursive directory iterator descends into directories and likely has some faulty logic which gets triggered on the CI. Furthermore recursion is not required.

### CODE REVIEW

**Comment by JehandadKhan** [comment]
waiting for the CI to see if this change has a positive impact on the issue.

File: src/find_db.cpp
Comment (junliume) [inline src/find_db.cpp:141]: @JehandadKhan so are we sure we will not have file structure like:
```
root_path
│   README.md
│   ASIC_1_db    
│
└───ASIC_2
│   │   ASIC_2_var1_db 
│   │   ASIC_2_var2_db 
│   │
│   └───ASIC_2_va3
│       │   ASIC_2_var3_bran1_db
│       │   ASIC_2_var3_bran2_db
│       │   ...
│   
```

Comment (JehandadKhan) [reply src/find_db.cpp:141]: That is correct.

**Comment by atamazov** [comment]
We can temporarily adjust Jenkinsfile to run this thread only on Vega20_64 and gfx1030/OCL -- to save resources and focus only on platforms that experience this issue. @JehandadKhan Please let me know if you wish me to do this.

**Comment by JehandadKhan** [comment]
> 
> 
> We can temporarily adjust Jenkinsfile to run this thread only on Vega20_64 and gfx1030/OCL -- to save resources and focus only on platforms that experience this issue. @JehandadKhan Please let me know if you wish me to do this.

I made that adjustment, please verify if it is correct, if not please update it to the correct config.

**Comment by atamazov** [comment]
> I made that adjustment, please verify if it is correct, if not please update it to the correct config.

Done. Also I've relabeled ixt-rack-54 for exclusive usage by this branch only.

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@junliume This PR contains some changes that needed to be rolled back before merging. I will prepare PR ASAP.

**Comment by atamazov** [comment]
> Done. Also I've relabeled ixt-rack-54 for exclusive usage by this branch only.

The node relabeled back and available for general use.

**Pull request merged**
